### DISCUSSION TITLE
Docker support for ARM processors?

### DISCUSSION THREAD

**Discussion by ParmuSingh** [JordanKnott/taskcafe#88] [Ideas]
I have had problems running docker instance on my raspberry pi 4. As it turns out the image is not for ARM architecture, which a rpi uses.

After going through some pain I was able to install from source and compile a binary for my rpi 

So the good news is that taskcafe does work on ARM.

The feature request is to have a docker image available for ARM architecture, as it'll help in easy deployment on rpi.

**Comment by JordanKnott** [comment]
If I am able to do it on my machine without actually having an ARM chip, then sure I can provide ARM images.

**Comment by ParmuSingh** [comment]
Anyway I can help with that? I compiled on a raspberry pi. If I share my binary would that help? I just don't know how to dockerize it.

**Comment by Fernthedev** [comment]
My best guess would be using an arm64 image of Ubuntu or some lightweight OS and run the same compile process for any Docker image (which I'm assuming is compiling the binary, and then building the image)
Of course, you'll have to make sure the arm version of the image uses everything ARM such as the OS and the GoLang tools for compiling (I'm assuming).

**Comment by JordanKnott** [comment]
I looked into it and from what I've found the best for me to release arm64 images (since I don't have an ARM based computer) is to use the `buildx` tool. Unfortunately it seems kind of annoying to install, but I'll try building the `0.4.0` release with it.

**Reply by Cyber1000** [reply]
@JordanKnott How to you create your docker-image? Manually? I couldn't find a .github/workflow-file ... 
I think the best would be to build with docker buildx, there is a github action for this, so no local install neccessary: https://github.com/docker/setup-buildx-action
I've done something similar before, but didn't need to push it to dockerhub.

**Reply by Cyber1000** [reply]
Something simple as this (without the multi-registry if not wanted), might suffice: https://github.com/docker/build-push-action/blob/master/docs/advanced/push-multi-registries.md

And the on-part could be something like this (executed on every creation of a new release, instead of every push to main):

```
on:
  release:
    types:
      - created
```

**Comment by ezcafe** [comment]
+1 for arm64 please

**Comment by mooujarrar** [comment]
I also want to run it on a raspberry pi which is running a linux armv7 can you pls provide an image compiled under this environment, thanks.

**Comment by ddjerqq** [comment]
I am a little late to the conversation, but if anyone else has an ARM CPU and wants to get this working:

1. Just clone the repo
2. and build the image on your ARM device. 

I got it working like that.

```bash
git clone https://github.com/JordanKnott/taskcafe.git
cd taskcafe
docker build . -t taskcafe:latest
docker run taskcafe:latest
```

That's all it takes!
